testsuite/model/src/test/java/org/keycloak/testsuite/model/infinispan/CacheExpirationTest.java

```diff
@@ -71,12 +74,16 @@ public void testCacheExpiration() throws Exception {
         // Ensure that instance counting works as expected, there should be at least two instances in memory now.
         // Infinispan server is decoding the client request before processing the request at the cache level,
         // therefore there are sometimes three instances of AuthenticationSessionAuthNoteUpdateEvent class in the memory
```

Comment:
@martin-kanis - I now see that when running on GitHub actions, when there are three instances in this location, there will be at the end one instance that will not be garbage collected. 

Looking at the git history, that these two lines were added by you. I wonder if you might recall where this one instance is kept, and the circumstances this occurs in. Even if you don't, it would be great if you could comment if it is ok for the test to succeed if this one instance is still on the heap.

Reply:
I discussed this problem with Ryan from Infinispan team. His exact explanation was:
"This is just a transient use of the class when the server is decoding the client request before processing the request at the cache level, hence why you only see it sometimes. I don't think this is anything to be concerned about."

I have a screenshot from heap dump which showing the references of this instance, if you are interested. 

So I don't think we need to worry about it in the test.

Reply:
Alright, I'll then re-write the test pass with one instance still in memory in the case when a third instance appeared after inserting the first two instances. When running the test on GitHub Actions, that one instance didn't disappear, it seems to be bound to some context so it can't be garbage collected. 

Please briefly confirm that you think this is the right way to go, and I'll then resolve this conversation.

Reply:
Sounds good